ykrt/ykrt.h

```diff
@@ -0,0 +1,55 @@
+#include <stdint.h>
+
+/// A meta-tracer.
+struct MT;
+typedef struct MT MT;
+
+// A `Location` stores state that the meta-tracer needs to identify hot loops
+// and run associated machine code.
+//
+// Each position in the end user's program that may be a control point (i.e.
+// the possible start of a trace) must have an associated `Location`. The
+// `Location` does not need to be at a stable address in memory and can be
+// freely moved.
+//
+// Program positions that can't be control points don't need an associated
+// `Location`. For interpreters that can't (or don't want) to be as selective,
+// a simple (if moderately wasteful) mechanism is for every bytecode or AST
+// node to have its own `Location` (even for bytecodes or nodes that can't be
+// control points).
+typedef struct {
+    uintptr_t state;
```

Comment:
Rust, unfortunately, conflates "the size of an int" with "the size of a pointer". Modern C differentiates them as `size_t` and `uintptr_t`: since `Location`s encode a pointer, the latter is the right thing for us.

Reply:
> I thought that Location may encode a pointer.

Yes... but I'm not sure what deeper question you're asking?

Reply:
Yeah, I was confused.